optimum/onnxruntime/modeling_seq2seq.py

```diff
@@ -0,0 +1,623 @@
+                standard cache should not be used.
+            force_download (`bool`, *optional*, defaults to `False`):
+                Whether or not to force the (re-)download of the model weights and configuration files, overriding the
+                cached versions if they exist.
+            encoder_file_name(`str`, *optional*):
+                The encoder model file name overwriting the default file name, allowing to save the encoder model with
+                a different name.
+            decoder_file_name(`str`, *optional*):
+                The decoder model file name overwriting the default file name, allowing to save the decoder model with
+                a different name.
+            decoder_with_past_file_name(`str`, *optional*):
+                The decoder with past key values model file name overwriting the default file name, allowing to save
+                the decoder model with a different name.
+            kwargs (`Dict`, *optional*):
+                kwargs will be passed to the model during initialization.
+        """
+        config_dict = kwargs.pop("config", {})
+        config = PretrainedConfig.from_dict(config_dict)
+        encoder_file_name = encoder_file_name or ONNX_ENCODER_NAME
+        decoder_file_name = decoder_file_name or ONNX_DECODER_NAME
+        decoder_with_past_file_name = decoder_with_past_file_name or ONNX_DECODER_WITH_PAST_NAME
+
+        # Load model from a local directory
+        if os.path.isdir(model_id):
+            model = cls.load_model(
+                encoder_path=os.path.join(model_id, encoder_file_name),
+                decoder_path=os.path.join(model_id, decoder_file_name),
+                decoder_with_past_path=os.path.join(model_id, decoder_with_past_file_name),
+                config=config,
+            )
+            kwargs["model_save_dir"] = Path(model_id)
+            kwargs["encoder"] = encoder_file_name
+            kwargs["decoder"] = decoder_file_name
+            kwargs["decoder_with_past"] = decoder_with_past_file_name
+        # Load model from hub
+        else:
+            default_file_names = [ONNX_ENCODER_NAME, ONNX_DECODER_NAME, ONNX_DECODER_WITH_PAST_NAME]
+            model_file_names = [encoder_file_name, decoder_file_name, decoder_with_past_file_name]
+            # Download the encoder, decoder and decoder_with_past forming the model
+            for file_name, default_file_name in zip(model_file_names, default_file_names):
+                model_cache_path = hf_hub_download(
+                    repo_id=model_id,
+                    filename=file_name,
+                    use_auth_token=use_auth_token,
+                    revision=revision,
+                    cache_dir=cache_dir,
+                    force_download=force_download,
+                )
+                kwargs[default_file_name.split(".")[0]] = Path(model_cache_path).name
```

Comment:
This needs to be adjusted then as well with probably `f"last_{default_file_name.split(".")[0]}_name"`